BoxContentSDK/BoxContentSDK/Clients/BoxContentClientServerAuthWrapper.m

```diff
@@ -0,0 +1,36 @@
+#import "BoxContentClientServerAuthWrapper.h"
+#import "BoxAbstractSession.h"
+
+@interface BoxContentClientServerAuthWrapper ()
+
+@property (nonnull, nonatomic, readonly, strong) ServerAuthFetchTokenBlock fetchTokenBlock;
+
+@end
+
+@implementation BoxContentClientServerAuthWrapper
+
+- (instancetype)initWithToken:(nullable NSString *)token
+                       userId:(nullable NSString *)userId
+                     userInfo:(nullable NSDictionary *)userInfo
+              fetchTokenBlock:(nonnull ServerAuthFetchTokenBlock)fetchTokenBlock
+{
+    if (self = [super init]) {
+        _userId = userId;
+        _userInfo = userInfo;
+        _fetchTokenBlock = fetchTokenBlock;
+        
+        _boxClient = [BOXContentClient clientForNewSession];
+        [_boxClient setAccessTokenDelegate:self];
+        
+        [_boxClient session].accessToken = token;
+    }
+    
+    return self;
+}
+
+- (void)fetchAccessTokenWithCompletion:(void (^)(NSString *, NSDate *, NSError *))completion
+{
+    _fetchTokenBlock(_userId, _userInfo, completion);
```

Comment:
I don't see a lot of code directly accessing ivars these days. Might want to use `self.fetchTokenBlock`